portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -136,9 +167,13 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
     // Package a history of comment objects, the total number of comments, the default number of
     // comments, the total number of pages, the current page ID, and whether the latest user
-    // input was dangerous and rejected, to JSON format.
+    // input was dangerous and rejected, whether the user input values of number of comments to 
+    // display & page ID are invalid, to JSON format.
     commentDataPackage commentData = new commentDataPackage(comments, totalComments, DEFAULT_MAX_COMMENTS,
-                                                            totalPages, currentPageId, isLatestInputDangerous);
+                                                            totalPages, currentPageId, invalidMaxComments,
```

Comment:
There's too many parameters being passed to construct construct CommentDataPackage. Think about  creating a custom type that could potentially encapsulate some of the parameters here. Feel free to ping me if you want to discuss this offline. Also DEFAULT_MAX_COMMENTS is a constant. This doesn't need to be a class member. You can make this public and access it from CommentDataPackage.

Reply:
Gotcha! I'll redesign this. Thanks!